ui/imports/shared/popups/ProfilePopup.qml

```diff
@@ -77,8 +77,10 @@ StatusDialog {
         // All this should be improved more, but for now we leave it like this.
         const contactDetails = Utils.getContactDetailsAsJson(publicKey);
 
+        isCurrentUser = popup.profileStore.pubkey === publicKey;
+
         userPublicKey = publicKey;
-        userDisplayName = contactDetails.displayName;
+        userDisplayName = isCurrentUser ? Qt.binding(() => { return popup.profileStore.displayName }) : contactDetails.displayName;
```

Comment:
Just wondering, why isn't this a regular binding at the top of the component? Because you need to call `Utils.getContactDetailsAsJson(publicKey)` first?